runtime/vm/jvminit.c

```diff
@@ -2006,18 +2006,18 @@ VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved)
 				IDATA argIndex8 = 0;
 				IDATA argIndex9 = 0;
 				BOOLEAN sharedClassDisabled = FALSE;
+#if defined(J9VM_OPT_PORTABLE_SHARED_CACHE)
 				BOOLEAN processPortableSharedCacheFlag = FALSE;
 
-#if (defined(J9VM_ARCH_X86) || defined(J9VM_ARCH_S390) || defined(J9VM_ARCH_POWER) || defined(J9VM_ARCH_AARCH64))
 				IDATA argIndexXXPortableSharedCache = 0;
 				IDATA argIndexXXNoPortableSharedCache = 0;
-#endif /* defined(J9VM_ARCH_X86) || defined(J9VM_ARCH_S390) || defined(J9VM_ARCH_POWER) || defined(J9VM_ARCH_AARCH64) */
 #if defined(J9VM_OPT_JITSERVER)
 				IDATA argIndexXXJITServerAOTCache = 0;
 				IDATA argIndexXXNoJITServerAOTCache = 0;
 				IDATA argIndexXXJITServerAOTCacheIgnoreLocalSCC = 0;
 				IDATA argIndexXXNoJITServerAOTCacheIgnoreLocalSCC = 0;
 #endif /* defined(J9VM_OPT_JITSERVER) */
+#endif /* defined(J9VM_OPT_PORTABLE_SHARED_CACHE) */
```

Comment:
Is it intentional that `J9VM_OPT_JITSERVER` is not only considered if `J9VM_OPT_PORTABLE_SHARED_CACHE`?
Before the code was structured so a configuration could support AOT caching, but not necessarily portable; that's no longer an option.

Reply:
We happen to parse that option twice - once here, and once in `J9Options`. We do it here just so that we can signal that we want to process the portable shared cache flag if it looks like the JVM is going to be a JITServer client.

Reply:
I don't see any _parsing_ of the portable shared cache options in `J9Options.cpp`, so it seems `jvminit.c` is the only place that calls `FIND_AND_CONSUME_VMARG()` for those options, thus my question remains: why do we want to change this so AOT cache options are only considered if `J9VM_OPT_PORTABLE_SHARED_CACHE`?

During my investigation, I noted that `shrinit.cpp` prints help for the portable shared cache options only on x86 (lines 1253 and 1259): `J9VM_OPT_PORTABLE_SHARED_CACHE` should also be used there (it doesn't currently apply on Windows x86 but the help would suggest otherwise).

Reply:
I was referring to the handling of the `-XX:[+|-]JITServerUseAOTCache` option here

https://github.com/eclipse-openj9/openj9/blob/383bb66f1dfbe1d2f429aa25bed1134206cdc4f2/runtime/compiler/control/J9Options.cpp#L1354-L1355

https://github.com/eclipse-openj9/openj9/blob/383bb66f1dfbe1d2f429aa25bed1134206cdc4f2/runtime/compiler/control/J9Options.cpp#L1419-L1422

I thought you were saying that because the `#if defined(J9VM_OPT_JITSERVER)` lines are now also guarded by the `#if defined(J9VM_OPT_PORTABLE_SHARED_CACHE)` in that part of `jvminit.c`, JVMs on platforms not supporting `J9VM_OPT_PORTABLE_SHARED_CACHE` would never have that JITServer AOT cache option parsed.

Reply:
> I thought you were saying that because the `#if defined(J9VM_OPT_JITSERVER)` lines are now also guarded by the `#if defined(J9VM_OPT_PORTABLE_SHARED_CACHE)` in that part of `jvminit.c`, JVMs on platforms not supporting `J9VM_OPT_PORTABLE_SHARED_CACHE` would never have that JITServer AOT cache option parsed.

Yes, that's what I was saying. Please explain why that change is appropriate, or restore the old separation of `J9VM_OPT_JITSERVER` from what is being replaced by `J9VM_OPT_PORTABLE_SHARED_CACHE`.

Reply:
The portable AOT flag processing should be guarded with `J9VM_OPT_PORTABLE_SHARED_CACHE` because it doesn't make sense to set that flag on platforms that don't support it - that's true at the client or the server. If that's the case, then we can skip all the JITServer AOT cache related option processing in that part of the file, because its only purpose there is to trigger the portable AOT flag processing.

This shouldn't change any actual behaviour, I think, because the JITServer is currently only supported on platforms that support portable AOT. The previous code had a latent bug, I suppose, which could have been triggered if we ever added JITServer support to a platform that didn't support portable AOT.

Reply:
Earlier I thought this change implied that AOT caching **_requires_** portable cache support (which I think is unreasonable), but after reading this again, I see that is not true (AOT caching options are handled in `shrclssup.c`). I think the code as proposed is correct.